Utils/Dataflow/pyDKB/common/misc.py

```diff
@@ -0,0 +1,49 @@
+"""
+pyDKB.common.misc
+
+Miscellanious utility functions.
+"""
+
+import sys
+import inspect
+from datetime import datetime
+
+from types import logLevel
+
+# Datetime format for log messages
+DTFORMAT = '%Y-%m-%d %H:%M:%S'
+
+
+def log(message, level=logLevel.INFO, *args):
+    """ Output log message with given log level.
+
+    :param message: message to output
+    :type message: str
+    :param level: log level of the message
+    :type level: ``pyDKB.common.types.logLevel`` member
+    :param *args: additional prefixes (will be output between log
+                  level prefix and message body)
+    :type *args: str
+    """
+    if not logLevel.hasMember(level):
+        log("Unknown log level: %s" % level, logLevel.WARN)
+        level = logLevel.INFO
+    if type(message) == list:
+        lines = message
+    else:
+        lines = message.splitlines()
+    if args:
+        prefix = ' ' + ' '.join(['(%s)' % p for p in args])
+    else:
+        frm = inspect.stack()[1]
+        mod = inspect.getmodule(frm[0])
+        modname = getattr(mod, '__name__', 'main')
+        prefix = ' (%s)' % modname
+    if lines:
+        dtime = datetime.now().strftime(DTFORMAT)
+        out_message = "%s (%s)%s %s" % (dtime, logLevel.memberName(level),
```

Comment:
Wouldn't it be better to keep the pre-message part of the strings the same length by adjusting log level format to their longest member? To replace current

    2019-08-06 15:48:54 (WARN) (__main__) 123
    2019-08-06 15:48:54 (FATAL) (__main__) 456

with something 'nicer', such as

    2019-08-06 15:48:54 (WARN)  (__main__) 123
    2019-08-06 15:48:54 (FATAL) (__main__) 456

?

Reply:
There may be somethig in this idea, but:
* broken alignment may make it easier for an eye to catch a changed pattern;
* the second prefix (here: `(__main__)`) is also not supposed to be aligned with others (which may be `(ProcessorStage)` or `(pyDKB)` or whatever, depending on the caller);
* when one parses the log automatically, it is easier if he/she can count on single space after the log level marker. Something like `\s+` will also work, of course, but why make things more difficult?

So I\`d rather leave it as is for now; if it will be annoying, it can always be changed later.

Reply:
Fair enough.